### ISSUE TITLE
Remove UTF-7 support for email bodies

### ISSUE THREAD

**Issue by pdg137** [mikel/mail#1618]
In the last year or two I have started to receive a few legitimate emails encoded as UTF-7 (maybe 1 or 2 per 100,000 emails). I noticed that despite attempting to support UTF-7 email bodies, the Mail gem does not decode these emails properly.

Note that standard UTF-7 uses a `+` character to initiate a special sequence (see [Wikipedia](https://en.wikipedia.org/wiki/UTF-7)). The version using `&`, which the Mail gem supports, appears to be the "modified UTF-7" described in [RFC 9051](https://datatracker.ietf.org/doc/html/rfc9051) and is only intended for email headers/addresses. It is certainly not a valid encoding for email bodies.

Note that

* Ruby does not support conversion to/from UTF-7:
  ```ruby
  irb(main):021> "Hi!".encode(Encoding::UTF_7)
  (irb):21:in `encode': code converter not found (UTF-8 to UTF-7) (Encoding::ConverterNotFoundError)
          from (irb):21:in `<main>'
  
  irb(main):022> "Hi+ACE-".force_encoding(Encoding::UTF_7).encode(Encoding::UTF_8)
  (irb):22:in `encode': code converter not found (UTF-7 to UTF-8) (Encoding::ConverterNotFoundError)
          from (irb):22:in `<main>'
  ```
* The Net::IMAP package only supports the "modified UTF-7" version in its [decode_utf7](https://apidock.com/ruby/Net/IMAP/decode_utf7/class) method.
* [Microsoft recommends against it](https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/utf-7-code-paths-obsolete).
* [The HTML5 standard](https://html.spec.whatwg.org/#character-encodings) disallows UTF-7 support.
* We have issues like this related to attempting the decoding: https://github.com/mikel/mail/issues/1404.

Given that we do not support the correct version of UTF-7 now, that Ruby itself does not support it, and there are many standards recommending against it, I think we should completely remove support for modified UTF-7 in email bodies.

**Comment by chadrschroeder** [comment]
I agree that it might be best for the Mail gem to stop supporting this obsolete encoding.  If an email uses UTF-7 and the email body includes text between `&` and `-` in a way that doesn't produce a valid character, that can cause a crash:

```ruby
mail = Mail.new do
  content_type 'text/html; charset="utf-7"'
  body '&nbsp; e-mail'
end
mail.decoded
# incomplete ")" on UTF-16BE (Encoding::InvalidByteSequenceError)
```

Here are some examples just calling the `decode_utf7` method directly:

```ruby
Mail::Utilities.decode_utf7('&AO0-')
 => "í"
Mail::Utilities.decode_utf7('&oops-')
# incomplete "l" on UTF-16BE (Encoding::InvalidByteSequenceError)
```

`Net::IMAP.decode_utf7` behaves similarly.

Also, `decode_utf7` doesn't handle UTF-7 text that encodes a character like "í" using "+AO0-" instead of "&AO0-":

```ruby
Mail::Utilities.decode_utf7('+AO0-')
 => "+AO0-"
```

But tools like `iconv` are able to recognize this format:

```
echo "+AO0-" >> test.txt
iconv -f utf-7 -t utf-8 test.txt
í
```
